raiden/transfer/channel.py

```diff
@@ -504,8 +506,24 @@ def is_valid_lock_expired(
 
     else:
         locksroot_without_lock = merkleroot(merkletree)
+        has_expired = is_lock_expired(
+            end_state=receiver_state,
+            lock=lock,
+            block_number=block_number,
+        )
 
-        if received_balance_proof.locksroot != locksroot_without_lock:
+        if not has_expired:
+            msg = (
+                "Invalid LockExpired message. "
```

Comment:
Since the `is_lock_expired()` function can actually tell us why it has not expired we can make the message here more correct. Can you change `is_lock_expired()` to return the reason it has not expired and use it in the message here?

Because if `is_lock_expired` returns false, due to the secret being registered on chain but the current block is still after the expiration of the lock this error message will be wrong as it will be saying something like: `expires at: 5 current block: 10.`